base/deprecated.jl

```diff
@@ -1151,6 +1151,19 @@ end
 isequal(x::Char, y::Integer) = false
 isequal(x::Integer, y::Char) = false
 
+function checkbounds(::Type{Bool}, sz::Integer, i)
+    depwarn("checkbounds(Bool, size(A, d), i) is deprecated, use checkindex(Bool, indices(A, d), i).", :checkbounds)
```

Comment:
is checkindex exported?

does this need revision of the bounds-checking NEWS.md entries?